test/CumulativeMerkle.test.js

```diff
@@ -102,13 +107,16 @@ describe('Cumulative Merkle Distribution', function () {
             const claimAccounts = proofAccounts.slice((proofAccounts.length / 2) * index, (proofAccounts.length / 2) * (index + 1))
             const claimAmounts = Array.from(claimAccounts).map((claimAccount, _) => ethers.BigNumber.from(dist.claims[claimAccount].amount))
             const claimProofs = Array.from(claimAccounts).map((claimAccount, _) => dist.claims[claimAccount].proof)
-            const claimStructs = Array.from(claimAccounts).map((claimAccount, index) => [claimAccount, claimAmounts[index], claimProofs[index]])
-            const prevBalances = await Promise.all(claimAccounts.map(async (claimAccount, _) => await token.balanceOf(claimAccount)))
-            const expBalances = Array.from(prevBalances).map((prevAmmount, index) =>  parseInt(prevAmmount + claimAmounts[index], 10))
+            const claimBeneficiaries = Array.from(claimAccounts).map((claimAccount, _) => dist.claims[claimAccount].beneficiary)
+            const claimStructs = Array.from(claimAccounts).map((claimAccount, index) => [claimAccount, claimBeneficiaries[index], claimAmounts[index], claimProofs[index]])
+            const prevBalances = await Promise.all(claimBeneficiaries.map(async (beneficiary, _) => await token.balanceOf(beneficiary)))
             await merkleDist.batchClaim(merkleRoot, claimStructs)
 
-            const afterBalancesHex = await Promise.all(claimAccounts.map(async (claimAccount, _) => await token.balanceOf(claimAccount)))
+            const afterBalancesHex = await Promise.all(claimBeneficiaries.map(async (beneficiary, _) => await token.balanceOf(beneficiary)))
             const afterBalances = Array.from(afterBalancesHex).map((afterAmmount, _) => parseInt(afterAmmount["_hex"], 16))
+            const additions = Object.fromEntries(claimBeneficiaries.filter(onlyUnique).map((i) => [i, 0]))
+            claimBeneficiaries.forEach((beneficiary, index) => {additions[beneficiary] += parseInt(claimAmounts[index], 10)})
+            const expBalances = Array.from(prevBalances).map((prevAmmount, index) => parseInt(prevAmmount + additions[claimBeneficiaries[index]], 10))
```

Comment:
Just to double-check, this is to compute the total amount a beneficiary is getting, since they may be receiving rewards from several stakes, correct?

BTW, what does `[i, 0]` mean in L117?

Reply:
Yes exactly. The tests are run for each stakingProvider, so we need to check what else that beneficiary has already been rewarded.

That line created a dictionary where the key is `beneficiary` and the value is `0`. It allows us to do `+= parseInt(claimAmounts[index], 10)` on the next line